tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -0,0 +1,136 @@
+"""Class definition for Category Four validator."""
+
+from .models import ParserErrorCategoryChoices
+from .util import get_rpt_month_year_list
+import logging
+
+logger = logging.getLogger(__name__)
+
+class CaseConsistencyValidator:
+    """Caches records of the same case and month to perform category four validation while actively parsing."""
```

Comment:
@raftmsohani is this actually caching records of the same case and month herein? or just case? we'd want case and month.

Reply:
Jan's follow up PRs introduce another data structure that does exactly that.